processor/blockprocessor/block_processor.go

```diff
@@ -38,12 +42,20 @@ func MakeProcessorWithLedger(l *ledger.Ledger, handler func(block *ledgercore.Va
 }
 
 // MakeProcessor creates a block processor
-func MakeProcessor(genesis *bookkeeping.Genesis, genesisBlock *bookkeeping.Block, datadir string, handler func(block *ledgercore.ValidatedBlock) error) (processor.Processor, error) {
+func MakeProcessor(genesis *bookkeeping.Genesis, genesisBlock *bookkeeping.Block, dbRound uint64, datadir string, handler func(block *ledgercore.ValidatedBlock) error) (processor.Processor, error) {
 	initState, err := util.CreateInitState(genesis, genesisBlock)
 	if err != nil {
 		return nil, fmt.Errorf("MakeProcessor() err: %w", err)
 	}
-	l, err := ledger.OpenLedger(logging.NewLogger(), filepath.Join(path.Dir(datadir), "ledger"), false, initState, algodConfig.GetDefaultLocal())
+	if dbRound != 0 && !ledgerExists(datadir, Prefix) {
+		msg := fmt.Sprintf("%s\n%s\n%s\n%s\n",
+			"The ledger cache was not found in the data directory and must be initialized. There are several ways to initialize it:",
+			"1.Fetch blocks and re-initialize, this takes a long time and is the most secure",
+			"2.Initialize with a catchpoint, this requires trusting that the relay is providing the correct ledger snapshot.",
+			fmt.Sprintf("3.Copy files ledger.block.sqlite and ledger.tracker.sqlite from an existing node installation from before round %d into the indexer data directory.", dbRound))
+		return nil, fmt.Errorf("MakeProcessor() err: %s", msg)
```

Comment:
I don't think it's needed for this PR, maybe not even this release, but we should find a better way to get this information to the user.